src/suggestionlistworker.h

```diff
@@ -0,0 +1,34 @@
+#ifndef SUGGESTIONLISTWORKER_H
+#define SUGGESTIONLISTWORKER_H
+
+#include <QObject>
+#include <QUrl>
+#include <QMutex>
+#include <QSemaphore>
+#include <QVector>
+
+class SuggestionListWorker : public QObject
+{
+    Q_OBJECT
+public:
+    explicit SuggestionListWorker(QObject *parent = nullptr);
+
+private:
+    bool checkAbort(const QString& text);
+
+public slots:
+    void searchSuggestions(const QString& text);
+    void setActualSearch(const QString& text);
+    bool isRunning() { return (m_semaphore.available() == 0); }
+
+signals:
+    void searchFinished(const QStringList& suggestions, const QVector<QUrl>& urlList);
+    void searchAborted();
+
+private:
+    QSemaphore m_semaphore;
+    QMutex m_mutex;
```

Comment:
I'm not sure about what the semaphore and the mutex are protecting.. and why two type of thread synchronization ?
Can you add some comments ? (And potentially rename them to a more explicit name)